### PR TITLE
Upgrade packages, rework test suite

### PR DESCRIPTION
**Pull request by euri10**
I've been frustrated by being forced to have a local postgres install just for the sake of running tests, so this is an attempt to solve https://github.com/nackjicholson/aiosql/issues/16

I made a few changes that I hope simplify the development for those who don't have a local db running on 5432.
For those who use one, it will be just a matter of using the POSTGRES_DNS env var to run them

summary of changes:

- removed psycopg2 for psycopg2-binary

- removed pytest-postgresql dependency and use a POSTGRES_DNS env var to point to the test db

- wrote a small test.sh script that enforces black, flake8 and runs the tests

- slightly modified travis.yml to test the lib on 3.6 to 3.8

- upgrade dependencies to latest versions

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,13 +1,22 @@
 import csv
+import logging
+import os
 import sqlite3
 from pathlib import Path
 
+import psycopg2
 import pytest
 
+assert "POSTGRES_DSN" in os.environ, "Postgres dsn not set"
```
Comment (xlevus) [inline tests/conftest.py:10]: Isn't this going to fail when `POSTGRES_DSN` is not set?

Comment (euri10) [reply tests/conftest.py:10]: > Isn't this going to cause the entire test suite to fail when `POSTGRES_DSN` is not set?


Yes that's the whole point of the pr, be more flexible, not everyone has postgresql running locally, currently the test suite fails if you're not running a local postgresql that listens on 5432. 

> Running `pytest` should work, and should not require any specific environment variables to be present.

It does work, the pr could default the env var to localhost 5432 if not set though.

**Comment by nackjicholson** [comment]
@euri10 This one looks like it needs some deeper review. I actually liked that pytest-postgresql managed the setup/teardown of the database.

As an alternative could we separate the test suite into `unit/` and `integration/` folders. The unit tests being things that don't need a PG running, then the integration tests do need a PG running.

**Comment by nackjicholson** [comment]
@euri10 Would this also have allowed you to point at a docker version of postgresql?

https://github.com/ClearcodeHQ/pytest-postgresql#connecting-to-already-existing-postgresql-database

pytest-postgresql also can easily be configured to point wherever you want via configuration settings.

https://github.com/ClearcodeHQ/pytest-postgresql#configuration

```sh
pytest --postgresql-port=5555 # wherever your instance of postgres is running (in docker)
```

With these things was there anything actually forcing you to have postgresql installed on your laptop locally? You could have used your throw-away docker instance I think.

**Comment by nackjicholson** [comment]
@euri10 Sorry about the conflicts. I liked some of the aspects of this and I merged them. I updated packages and incorporated your build changes.

I'm still not ready to merge the test rework. I'm not all the way convinced it was necessary, but I'm also willing to admit I could be wrong about that :stuck_out_tongue_closed_eyes: :

**Comment by nackjicholson** [comment]
@xlevus @euri10 I did this differently without the required environment variable in #40. When you do have `postgres` installed locally the tests can work exactly the same and still use `pytest-postgresql`. For the case where someone wants to run tests against a detached instance of postgres (like a docker container or RDS) you can use the [pytest-postgresql CLI options](https://github.com/ClearcodeHQ/pytest-postgresql#configuration) to configure the connection to the remote instance along with a new option I added to aiosql tests `--postgresql-detached`.

**Pull request closed without merging**